Service/Request/RequestFactory.php

```diff
@@ -149,6 +147,26 @@ private function getRequestUri(ServiceRequestInterface $serviceRequest): UriInte
         return $this->uriFactory->createUri($baseUrl.$path);
     }
 
+    /**
+     * @param ServiceRequestInterface $serviceRequest
+     * @param EndpointInterface $endpoint
+     * @return mixed
+     */
+    private function getRequestBody(ServiceRequestInterface $serviceRequest, EndpointInterface $endpoint)
+    {
+        $isMethodWithoutBody = in_array($endpoint->getMethod(), self::METHODS_WITHOUT_BODY, true);
+
+        if ($isMethodWithoutBody) {
+            $requestBody = null;
+        } elseif ($serviceRequest instanceof StreamInterface) {
+            $requestBody = $serviceRequest;
+        } else  {
+            $requestBody = $this->serializer->serialize($serviceRequest, $endpoint->getRequestFormat());
+        }
+
+        return $requestBody;
```

Comment:
We can do early return, instead of setting to `$requestBody`

Reply:
Fixed

Reply:
@justenj I meant we could return early for all those `$requestBody` variables, we don't need `$requestBody` at all.

```php
if ($this->isMethodWithoutBody($endpoint->getMethod()) && $this->strictModeEnabled) {
  return null;
}

if ($serviceRequest instanceof StreamInterface) {
  return $serviceRequest;
}

return $this->serializer->serialize($serviceRequest, $endpoint->getRequestFormat());
```